packages/client/hmi-client/src/components/widgets/tera-infinite-canvas.vue

```diff
@@ -0,0 +1,162 @@
+<template>
+	<main ref="canvasRef">
+		<div>
+			<slot name="foreground" />
+		</div>
+		<div class="data-layer" ref="dataLayerRef">
+			<slot name="data" />
+		</div>
+		<svg class="background-layer" ref="backgroundLayerRef" :width="width" :height="height">
+			<slot name="background" />
+		</svg>
+	</main>
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted, onUnmounted } from 'vue';
+import * as d3 from 'd3';
+
+const props = withDefaults(
+	defineProps<{
+		debugMode?: boolean;
+		scaleExtent?: [number, number];
+		lastTransform?: { k: number; x: number; y: number };
+	}>(),
+	{
+		debugMode: false,
+		scaleExtent: () => [0.1, 10],
```

Comment:
We can just return `[0.1, 10]` directly not not wrap it in a function?

Reply:
I did that because I get this type issue if I don't. 
![image](https://user-images.githubusercontent.com/28237258/234883489-58b2dc34-f013-4308-8958-7df78405bdb9.png)

In the docs if it's an array they make it a function: https://vuejs.org/guide/typescript/composition-api.html#typing-component-props

There may be a way of setting up the props in the runtime way which is the very first example on that page. I just had trouble declaring proper typing like `[Number, Number]`

Reply:
okay, let's not worry about this now.